### DISCUSSION TITLE
Why use AsyncSemaphore in subscribe(long threadId) and unsubscribe(RedissonLockEntry entry, long threadId) ?

### DISCUSSION THREAD

**Discussion by dk900912** [redisson/redisson#4903] [General]
```java
        AsyncSemaphore asyncSemaphore = new AsyncSemaphore(1);
        asyncSemaphore.acquire().thenAccept(future -> {
            bizAction();
            asyncSemaphore.release();
        });
```
In my opinion, this seems to be for traffic limiting in multi-thread mode, why we need this for lock() ?

**Comment by mrniko** [comment]
Some actions can't be executed concurrently.

**Reply by dk900912** [reply]
Hi，mrniko. That is to say, there will be problems when multiple threads subscribe or unsubscribe redisson_lock__channel:{my-lock} at the same time?

**Reply by mrniko** [reply]
No, since these methods are guarded by AsyncSempaphore

**Reply by dk900912** [reply]
thank u，niko
